block/internal/syncing/syncer.go

```diff
@@ -391,7 +376,7 @@ func (s *Syncer) processHeightEvent(event *common.DAHeightEvent) {
 
 	// Try to sync the next block
 	if err := s.trySyncNextBlock(event.DaHeight); err != nil {
-		s.logger.Error().Err(err).Msg("failed to sync next block")
+		s.errorCh <- fmt.Errorf("failed to sync next block: %w", err)
```

Comment:
we do not want to halt here, we may have received some gibberish invalid header. we do the signature verification in trySyncNextBlock, as we need block n-1 executed to do so.